events/apigw.go

```diff
@@ -65,17 +65,18 @@ type APIGatewayV2HTTPRequest struct {
 
 // APIGatewayV2HTTPRequestContext contains the information to identify the AWS account and resources invoking the Lambda function.
 type APIGatewayV2HTTPRequestContext struct {
-	RouteKey     string                                               `json:"routeKey"`
-	AccountID    string                                               `json:"accountId"`
-	Stage        string                                               `json:"stage"`
-	RequestID    string                                               `json:"requestId"`
-	Authorizer   *APIGatewayV2HTTPRequestContextAuthorizerDescription `json:"authorizer,omitempty"`
-	APIID        string                                               `json:"apiId"` // The API Gateway HTTP API Id
-	DomainName   string                                               `json:"domainName"`
-	DomainPrefix string                                               `json:"domainPrefix"`
-	Time         string                                               `json:"time"`
-	TimeEpoch    int64                                                `json:"timeEpoch"`
-	HTTP         APIGatewayV2HTTPRequestContextHTTPDescription        `json:"http"`
+	RouteKey       string                                               `json:"routeKey"`
+	AccountID      string                                               `json:"accountId"`
+	Stage          string                                               `json:"stage"`
+	RequestID      string                                               `json:"requestId"`
+	Authorizer     *APIGatewayV2HTTPRequestContextAuthorizerDescription `json:"authorizer,omitempty"`
+	APIID          string                                               `json:"apiId"` // The API Gateway HTTP API Id
+	DomainName     string                                               `json:"domainName"`
+	DomainPrefix   string                                               `json:"domainPrefix"`
+	Time           string                                               `json:"time"`
+	TimeEpoch      int64                                                `json:"timeEpoch"`
+	HTTP           APIGatewayV2HTTPRequestContextHTTPDescription        `json:"http"`
+	Authentication APIGatewayV2HTTPRequestContextAuthentication         `json:"authentication"`
```

Comment:
This field is optional right? Or is it always supplied now by API GW?

Reply:
Supplied by API GW when mTLS is configured, otherwise it will be ignored since we are just unmarshalling.